File: 프로그래머스/lv1/12943. 콜라츠 추측/콜라츠 추측.js
function solution(num) {

  if (num === 1) { return 0 }

  let n = 0
  while (num !== 1) {
    if (num % 2 === 0) { num = num/2 }
    else { num = num*3+1 }
    n ++ 
  }

  return n < 500 ? n : -1
}

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.py
def solution(num):
    if num == 1: return 0
    cnt = 0
    while 1:
        num = num//2 if (num%2 == 0) else num*3+1
        cnt += 1
        if num == 1: return cnt
        if cnt == 500: return -1
    


print(solution(6))